dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,649 @@
+              range.size() == 2 ? range[1] : -1.0f) {
+  }
+
+  bool Contains(float k) const {
+    return k >= min && k <= max;
+  }
+
+  const float min, max;
+};
+
+}  // namespace
+
 DALI_SCHEMA(RandomBBoxCrop)
     .DocStr(
-        R"code(Perform a prospective crop to an image while keeping bounding boxes and labels consistent. Inputs must be supplied as
-two Tensors: `BBoxes` containing bounding boxes represented as `[l,t,r,b]` or `[x,y,w,h]`, and `Labels` containing the
-corresponding label for each bounding box. Resulting prospective crop is provided as two Tensors: `Begin` containing the starting
-coordinates for the `crop` in `(x,y)` format, and 'Size' containing the dimensions of the `crop` in `(w,h)` format.
-Bounding boxes are provided as a `(m*4)` Tensor, where each bounding box is represented as `[l,t,r,b]` or `[x,y,w,h]`. Resulting
-labels match the boxes that remain, after being discarded with respect to the minimum accepted intersection threshold.
-Be advised, when `allow_no_crop` is `false` and `thresholds` does not contain `0` it is good to increase `num_attempts` as otherwise
-it may loop for a very long time.)code")
-    .NumInput(2)
-    .NumOutput(4)
+        R"code(Performs a prospective random crop to an image coordinate space while keeping
+bounding boxes (and optionally labels) consistent. That is, after applying the random crop operator
+to the image coordinate space, the bounding boxes will be adjusted or filtered out to match the
+cropped region of interest. The applied random crop operator is constrained by the arguments
+provided to the operator.
+
+The cropping window random selection is selected at random until it matches the overlap restrictions
+specified by `thresholds` argument. Thresholds values represent minimum intersection-over-union
+(IoU) of the cropping window and each of the provided bounding boxes. A valid crop is the one where
+all the provided bounding boxes produce an IoU above a give value present in `thresholds` randomly
+selected. Alternative, we can allow not to crop as one of the valid outcomes of the random process
+(use `allow_no_crop` for that).
+
+Two modes of random crop are available:
+
+* Randomly shaped window, randomly placed within the original input space. The random crop window
+dimensions are selected according to the provided `aspect_ratio` restrictions.
+
+* Fixed size window, randomly placed within the original input space. The random crop window
+dimensions is fixed to match the provided `crop_shape` argument and the anchor is selected randomly.
+When providing `crop_shape`, a second argument `input_shape` should be provided, specifying the
+original image dimensions.
+
+The argument `num_attempts` can be used to control the maximum number of attempts that we try to
+produce a valid crop to match a single minimum IoU value from `thresholds`. Be advised that when
+`allow_no_crop` is `False` and `thresholds` does not contain `0.0`, it is good to increase the
```

Comment:
Increase or decrease. If it is very likely that there is no valid crop for given image then only decreasing `num_attempts` would help speed things up.

Reply:
decrease actually makes you change to a different threshold earlier, so that might be why it was written that way. The logic is as follows:

```
while not a valid crop:
  chose threshold
  for attemtps < num_attempts:
     crop = random crop with threshold
```

Reply:
Now I understand. I though that `for attemtps < num_attempts:` was the outermost loop.